src/components/groups/CreateGroup.tsx

```diff
@@ -0,0 +1,191 @@
+import React from "react";
+import { useState } from "react";
+import { useCreateGroupMutation } from "<@>/store/groups/groups-api";
+import Group from "<@>/types/groups/group-body";
+
+interface CreateGroup {
+  handleClose: () => void;
+}
+
+const initialState: Group = {
+  name: "",
+  capacity: 0,
+  telegramLink: "",
+  divisionId: 1,
+};
+const initialError = {
+  name: "",
+  capacity: "",
+  telegramLink: "",
+  divisionId: "",
+};
+
+const CreateGroup: React.FC<CreateGroup> = ({ handleClose }) => {
+  const [group, setGroup] = useState(initialState);
+  const [createData, { isLoading, error }] = useCreateGroupMutation();
+  const [errors, setErrors] = useState(initialError);
+  const [onCreate, setCreat] = useState<boolean>(false);
+
+  const handleChange = (event: any) => {
+    const { name, value } = event.target;
+    setGroup((prevErrors) => ({
+      ...prevErrors,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    const { name, capacity, telegramLink, divisionId } = group;
+
+    if (!name) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        name: "group name is required",
+      }));
+      return;
+    }
+
+    if (!capacity) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        capacity: "group size is required",
+      }));
+      return;
+    }
+
+    if (!telegramLink) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        telegramLink: "telegram link is required",
+      }));
+      return;
+    }
+
+    if (!divisionId) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        divisionId: "division level is required",
+      }));
+      return;
+    }
+
+    try {
+      await createData({
+        name,
+        capacity: capacity as number,
+        telegramLink,
+        divisionId: divisionId as number,
+      }).unwrap();
+      setGroup(initialState);
+      setErrors(initialError);
+    } catch (errors) {}
```

Comment:
If you find that a try-catch block is important and you need to do something after catching an error, perform the necessary actions in the catch block